src/external_acl.cc

```diff
@@ -915,6 +916,7 @@ class externalAclState
     external_acl *def;
     dlink_node list;
     externalAclState *queue;
+    CodeContextPointer codeContext;
 };
```

Comment:
Please check whether we can avoid adding custom CodeContext handling code by converting/upgrading the existing callback/callback_data pair into an async call. I would not be surprised if cbdataDialer() can be used for this, similar to what helperOpenServers() does today. AFAICT, callback_data is always an ACL Checklist here, but I am not sure that knowledge can be used to simplify anything.

We could probably reusing AsyncCall queuing to queue externalAclState objects as well, but I would not go there (in this PR or at all) because externalAclState objects are also used for external ACL lookup results _caching_, and that may complicate things quite a bit.